### PR TITLE
returning assets url in graphql. Moving AssetIdentifier logic to indexer-core

### PR DESCRIPTION
**Pull request by mpwsh**
- moved `AssetIdentifier` from `indexer/http/metadata_json` to `indexer-core/assets.rs`
- added function to return detected service and Cid in `AssetsIdentifier`
- implemented transforming url to `assets.holaplex.com` in graphql

### CODE REVIEW

**Review by ray-kast — CHANGES_REQUESTED** [review]
I would prefer the logic for assembling a CDN link to be handled by the GraphQL server rather than the `get_cid_with_svc` function.

File: crates/graphql/src/schema/query_root.rs [hunk 1]
```diff
@@ -177,7 +176,14 @@ impl QueryRoot {
             .load(&conn)
             .context("Failed to load metadata")?;
 
-        Ok(rows.pop().map(Into::into))
+        Ok(rows.pop().map(|mut n| {
```
Comment (kespinola) [inline crates/graphql/src/schema/query_root.rs:179]: Its cleaner do this swap in the image method of the nft object instead of in the dataloader.

**Review by kespinola — CHANGES_REQUESTED** [review]

File: crates/graphql/src/schema/query_root.rs (hunk 1 above)
Comment (mpwsh) [reply crates/graphql/src/schema/query_root.rs]: Added proposed changes in image method

File: crates/graphql/src/schema/objects/nft.rs [hunk 2]
```diff
@@ -158,8 +161,20 @@ impl Nft {
         &self.description
     }
 
-    pub fn image(&self) -> &str {
-        &self.image
+    pub fn image(&self) -> String {
+        let assets_cdn = "https://assets.holaplex.com";
+        let asset = AssetIdentifier::new(&Url::parse(&self.image).unwrap());
```
Comment (ray-kast) [inline crates/graphql/src/schema/objects/nft.rs:166]: Should check if both fields are Some.  It is technically not impossible that a link is ambiguous, in which case I think returning the raw URL is not out of the question.

**Review by ray-kast — CHANGES_REQUESTED** [review]

File: crates/graphql/src/schema/objects/nft.rs [hunk 3]
```diff
@@ -158,8 +161,20 @@ impl Nft {
         &self.description
     }
 
-    pub fn image(&self) -> &str {
-        &self.image
+    pub fn image(&self) -> String {
+        let assets_cdn = "https://assets.holaplex.com";
```
Comment (kespinola) [inline crates/graphql/src/schema/objects/nft.rs:165]: Configure the asset host in later PR.

**Review by kespinola — APPROVED** [review]

File: crates/graphql/src/schema/objects/nft.rs (hunk 3 above)
Comment (ray-kast) [reply crates/graphql/src/schema/objects/nft.rs:165]: https://github.com/holaplex/solana-indexer/issues/227

**Review by ray-kast — APPROVED** [review]

**Pull request merged**